ibis/expr/operations/core.py

```diff
@@ -153,9 +152,12 @@ def is_ancestor(self, other):
         return self.equals(other)
 
     def to_expr(self):
-        if not hasattr(self, '_expr_cached'):
-            self._expr_cached = self._make_expr()
-        return self._expr_cached
+        try:
+            result = self._expr_cached
+        except AttributeError:
+            result = self._make_expr()
+            object.__setattr__(self, "_expr_cached", result)
```

Comment:
Shall we remove `_expr_cached` in this PR?

Reply:
Let's remove it in a follow up.